src/transformStateWithClones.js

```diff
@@ -7,7 +7,28 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  let currentState = { ...state };
+
+  for (const action of actions) {
+    const { type } = action;
+
+    if (type === 'addProperties') {
+      Object.assign(currentState, action.extraData);
+    }
+
+    if (type === 'removeProperties') {
+      action.keysToRemove.forEach(key => delete currentState[key]);
+    }
+
+    if (type === 'clear') {
+      currentState = {};
+    }
```

Comment:
better use a `switch..case` construct that provides you with a `default` case, where you can provide an explanatory fallback, like
```
throw new Error(`${action} is not supported`)
```